db/views/latest_blood_pressures_per_patient_per_quarters_v01.sql

```diff
@@ -0,0 +1,3 @@
+SELECT DISTINCT ON (patient_id, year, quarter) *
+FROM latest_blood_pressures_per_patient_per_months
+ORDER BY patient_id, year, quarter, bp_recorded_at DESC, bp_id;
```

Comment:
Is it worth introducing the complexity of a view derived from another view? Does the performance gain justify this? How terrible would it be for us to craft this view the same way we do the per-month table above, and simply `DISTINCT` on quarter instead? That would avoid the need to be careful about ordering when refreshing these views, and reduce the coupling between them.

As a sanity check, do we _need_ a per-quarter view, if we're already adding a `quarter` column to the per-month view?

Reply:
We do need the per quarter view, primarily to get the latest bp per quarter.

As for the derived views, we'll need to benchmark the view refresh against prod sized data, before we can conclusively decide. We'd also be extracting the month, quarter and year again for each view.

Maybe we can test this at some point and make the views independent then.

Reply:
Can we test with a large enough dataset _before_ we ship this? Is it hard to backtrack if we ship this as is and refresh the views in the wrong order?

Reply:
I thought about this a little more. I think the regular use of a the carefully crafted rake task (with associated comment) should cover us well. Let's punt on any performance testing for now.